wwudora.module

```diff
@@ -14,3 +14,61 @@ function wwudora_islandora_oai_get_xsl_files() {
     "$module_path/xml/wwu_mods_to_dc.xsl" => 'wwu_mods_to_dc.xsl',
   );
 }
+
+/**
+ * Implements hook_islandora_datastream_ingested().
+ */
+function wwudora_islandora_datastream_ingested($object, $datastream) {
+  wwudora_inherit_xacml_from_compound_parent($object, $datastream);
+}
+
+/**
+ * Implements hook_islandora_datastream_modified().
+ */
+function wwudora_islandora_datastream_modified($object, $datastream, $params) {
+  wwudora_inherit_xacml_from_compound_parent($object, $datastream);
+}
+
+/**
+ * Implements hook_islandora_compound_object_children_added_to_parent().
+ */
+function wwudora_islandora_compound_object_children_added_to_parent($children, $parent_pids) {
+  $parent = islandora_object_load(current($parent_pids));
+  foreach ($children as $child) {
+    $child->relationships->remove(ISLANDORA_RELS_EXT_URI, 'inheritXacmlFrom', NULL, RELS_TYPE_URI);
+    $child->relationships->add(ISLANDORA_RELS_EXT_URI, 'inheritXacmlFrom', $parent->id, RELS_TYPE_URI);
+    if ($parent['POLICY']) {
+      $xacml = new IslandoraXacml($child, $parent['POLICY']->content);
+      $xacml->writeBackToFedora();
+    }
+    elseif ($child['POLICY']) {
+      // Removing Child's existing Policy if new Parent doesn't have one.
+      $xacml = new IslandoraXacml($child);
+      $xacml->datastreamRule->clear();
+      $xacml->managementRule->clear();
+      $xacml->viewingRule->clear();
+      $xacml->writeBackToFedora();
+    }
+  }
+}
+
+/**
+ * Apply compound object policies to children if is ingested or modified.
+ *
+ * @param AbstractObject $object
+ *   An AbstractObject of the parent object.
+ * @param AbstractDatastream $datastream
+ *   An AbstractDatastream of the datastream added to the parent object.
+ */
+function wwudora_inherit_xacml_from_compound_parent(AbstractObject $object, AbstractDatastream $datastream) {
```

Comment:
This should probably go into a separate `.inc` file and get included by the hook implementations